pkg/scanners/trivy/types.go

```diff
@@ -143,6 +146,22 @@ func (c *Config) cliArgs(ref string) []string {
 	return args
 }
 
+func (c *Config) getRuntimeVar() string {
+	var imgsrc string
+	runtimeName := c.Runtime.Name
+
+	switch runtimeName {
+	case unversioned.RuntimeCrio:
+		imgsrc = ImgSrcPodman
+	case unversioned.RuntimeDockerShim:
+		imgsrc = ImgSrcDocker
+	default:
+		imgsrc = ImgSrcContainerd
+	}
+
+	return imgsrc
+}
+
```

Comment:
This has a bug in it. If the runtime name is "foo", it will get set to `containerd`. It should give an error in that case.

Try:
```suggestion
func (c *Config) getRuntimeVar() (string, error) {
	var imgsrc string
	runtimeName := c.Runtime.Name

	switch runtimeName {
	case unversioned.RuntimeCrio:
		imgsrc = ImgSrcPodman
	case unversioned.RuntimeDockerShim:
		imgsrc = ImgSrcDocker
        case unversioned.RuntimeContainerd, unversioned.Runtime(""):
                imgsrc = ImgSrcContainerd
	default:
		return "", fmt.Errorf("invalid runtime provided: %q", runtimeName)
	}

	return imgsrc, nil
}

```

Reply:
Incidentally, the bug is why setting `c.Runtime.Name` to `ERASER_RUNTIME_NAME` is currently working.